master/internal/agent/agents.go

```diff
@@ -38,6 +40,30 @@ func (a *agents) Receive(ctx *actor.Context) error {
 			ctx.Respond(errors.Errorf("agent already connected: %s", id))
 			return nil
 		}
+	case *proto.GetAgentsRequest:
+		response := &proto.GetAgentsResponse{}
+		for _, a := range a.summarize(ctx) {
+			if msg.Label != "" && msg.Label != a.Label {
+				continue
+			}
+			response.Agents = append(response.Agents, toProtoAgent(a))
```

Comment:
non-blocking: curious why not do this the other way:

```
if msg.Label == "" || msg.Label == a.Label {
    response.Agents = append(response.Agents, toProtoAgent(a))
}
```

I personally think this is a bit clearer, but it definitely is a personal preference.